streamer/src/nonblocking/quic.rs

```diff
@@ -237,33 +237,39 @@ async fn handle_connection(
                     Ok(mut stream) => {
                         stats.total_streams.fetch_add(1, Ordering::Relaxed);
                         stats.total_new_streams.fetch_add(1, Ordering::Relaxed);
-                        let mut maybe_batch = None;
-                        while !stream_exit.load(Ordering::Relaxed) {
-                            if let Ok(chunk) = tokio::time::timeout(
-                                Duration::from_millis(WAIT_FOR_STREAM_TIMEOUT_MS),
-                                stream.read_chunk(PACKET_DATA_SIZE, false),
-                            )
-                            .await
-                            {
-                                if handle_chunk(
-                                    &chunk,
-                                    &mut maybe_batch,
-                                    &remote_addr,
-                                    &packet_sender,
-                                    stats.clone(),
-                                    stake,
-                                ) {
-                                    last_update.store(timing::timestamp(), Ordering::Relaxed);
-                                    break;
+                        let stream_exit = stream_exit.clone();
+                        let stats = stats.clone();
+                        let packet_sender = packet_sender.clone();
+                        let last_update = last_update.clone();
+                        tokio::spawn(async move {
```

Comment:
The concern probably is we may spawn too many async tasks into the runtime in the server side compared with before. But I think that might be addressed by tweaking the maximum streams allowed for each connection if we find issues.